File: Python/beginner/1176.py
# https://judge.beecrowd.com/pt/problems/view/1176
# Fibonacci em Vetor

# Pre-calculating Fibonacci numbers up to F(60)
pyFibz = [0, 1]
for i in range(2, 61):
    pyFibz.append(pyFibz[i - 1] + pyFibz[i - 2])

# Asking for the number of test cases
testCases = int(input())

# Processing indexes and printing results
for _ in range(testCases):
    indexes = int(input())
    print(f"Fib({indexes}) = {pyFibz[indexes]}")
    
# By: Fernando Serra
# https://github.com/fernandosserra

File: C/beginner/1176.c
// https://judge.beecrowd.com/pt/problems/view/1176
// Fibonacci em Vetor

// Library
#include <stdio.h>

// Main function
int main(){
    // Variables
    int usrInput1, usrInput2;
    long long int fib[61];

    // Initialize the first two Fibonacci numbers
    fib[0] = 0;
    fib[1] = 1;

    // Calculate Fibonacci numbers up to the 59th
    for (int i = 2; i <= 60; i++){
        fib[i] = fib[i - 1] + fib[i - 2];
    }

    // Getting test cases
    scanf("%d", &usrInput1);

    // Loop through each test case
    for (int i = 0; i < usrInput1; i++){
        // Read the index for Fibonacci number
        scanf("%d", &usrInput2);
        
        // Print the Fibonacci number at the given index
        printf("Fib(%d) = %lld\n", usrInput2, fib[usrInput2]);
    }
    return 0;
}

// By: Fernando Serra
// https://github.com/fernandosserra